tests/e2e/helpers/workload.go

```diff
@@ -0,0 +1,151 @@
+/**
+# Copyright (c) NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package helpers
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+type WorkloadClient struct {
+	client kubernetes.Interface
+}
+
+func NewWorkloadClient(client kubernetes.Interface) *WorkloadClient {
+	return &WorkloadClient{
+		client: client,
+	}
+}
+
+func (h *WorkloadClient) DeployGPUPod(ctx context.Context, namespace string, podSpec *corev1.Pod) (*corev1.Pod, error) {
```

Comment:
nit: Technically the pod passed in does not need to be a GPU workload. Would renaming this function to `DeployPod` be a better fit?